go-controller/pkg/mock/mock_ovn.go

```diff
@@ -0,0 +1,469 @@
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Add chassis with given name
+func (mock *MockOVNClient) ChassisAdd(name string, hostname string, etype []string, ip string, external_ids map[string]string,
+	transport_zones []string, vtep_lswitches []string) (util.OVNCommandInterface, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Delete chassis with given name
+func (mock *MockOVNClient) ChassisDel(chName string) (util.OVNCommandInterface, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Get chassis by hostname or name
+func (mock *MockOVNClient) ChassisGet(chname string) ([]*goovn.Chassis, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Get encaps by chassis name
+func (mock *MockOVNClient) EncapList(chname string) ([]*goovn.Encap, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Set NB_Global table options
+func (mock *MockOVNClient) NBGlobalSetOptions(options map[string]string) (util.OVNCommandInterface, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Get NB_Global table options
+func (mock *MockOVNClient) NBGlobalGetOptions() (map[string]string, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Set SB_Global table options
+func (mock *MockOVNClient) SBGlobalSetOptions(options map[string]string) (util.OVNCommandInterface, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Get SB_Global table options
+func (mock *MockOVNClient) SBGlobalGetOptions() (map[string]string, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Close connection to OVN
+func (mock *MockOVNClient) Close() error {
+	return nil
+}
+
+// Exec command, support multiple commands in one transaction.
+// executes commands ensuring their temporal consistency
+func (mock *MockOVNClient) Execute(cmds ...util.OVNCommandInterface) error {
+	errors := make([]error, 0, len(cmds))
+	for _, cmd := range cmds {
+		// go over each mock command and apply the
+		// individual command's operations to the
+		// cache
+		ovnCmd, ok := cmd.(*MockOVNCommand)
+		if !ok {
+			klog.Errorf("type assertion failed for mock command")
+			errors = append(errors, fmt.Errorf("type assertion failed for mock command"))
+		}
+		var cache MockObjectCacheByName
+		switch ovnCmd.op {
+		case OpAdd:
+			if cache, ok = mock.cache[ovnCmd.table]; !ok {
+				cache = make(MockObjectCacheByName)
+				mock.cache[ovnCmd.table] = cache
+			}
+			if _, exists := cache[ovnCmd.objName]; exists {
+				errors = append(errors,
+					fmt.Errorf("object %s of type %s exists in cache", ovnCmd.objName, ovnCmd.table))
+			}
+			cache[ovnCmd.objName] = ovnCmd.obj
+		case OpDelete:
```

Comment:
Is this an error, or are deletions idempotent?

Reply:
That's up to us. Right now I am treating deleting a non-existent entry as error. That is perhaps a requirement that can be relaxed if we make deletes idempotent.

Reply:
What does the underlying library do? We should match that.

Reply:
Underlying library doesn't have the `--if-exists` semantics afaict so will follow the ovsdb delete transaction semantics. if an entry doesn't exist, it will be an error.